File: action-example/src/main/java/move/action/MyWorker.java
package move.action;

import javax.inject.Inject;

/**
 *
 */
@WorkerAction(fifo = true)
public class MyWorker extends AbstractWorkerAction<MyWorker.Request> {
    @Inject
    MyWorker() {
    }

    @Override
    protected void start(Request request) {
        System.out.println("MyWorker: " + System.currentTimeMillis());
        processed();
    }

    public static class Request {
        @Inject
        public Request() {
        }
    }
}

File: action-example/src/main/java/move/action/MyWorker.kt
package move.action

import kotlinx.coroutines.experimental.delay
import javax.inject.Inject

/**

 */
@WorkerAction(fifo = false)
class MyWorker @Inject constructor() : BaseWorkerAction<MyWorker.Request>() {
   suspend override fun recover(caught: Throwable, cause: Throwable, isFallback: Boolean): Boolean {
      return false
   }

   suspend override fun execute(): Boolean {
      println("Started worker")
//      delay(1000)
      println("Finishing worker")
      return true
   }

   class Request @Inject constructor() {
      var id: String? = null
   }
}
